eftest/src/eftest/report/junit.clj

```diff
@@ -93,25 +94,37 @@
     (println "</testsuites>")))
 
 (defmethod report :begin-test-ns [m]
-  (test/with-test-out
-    (start-suite (name (ns-name (:ns m))))))
+  (let [ns (name (ns-name (:ns m)))
+        f  #(test/with-test-out
+              (start-suite ns))]
+    (swap! *context* assoc-in [::deferred-report ns] f)))
 
 (defmethod report :end-test-ns [m]
-  (test/with-test-out
-    (finish-suite)))
+  (let [ns (name (ns-name (:ns m)))
+        f  #(test/with-test-out
+              (finish-suite))
+        g  (get-in @*context* [::deferred-report ns])]
```

Comment:
It might make more sense for `f` and `g` to be reversed, since `g` is executed before `f`.